sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/MetadataReferences/MetadataReferenceFacade.cs

```diff
@@ -0,0 +1,293 @@
+            };
+#endif
+
+        internal static IEnumerable<MetadataReference> GetSystemData() =>
+#if NETFRAMEWORK
+            FrameworkMetadataReference.SystemData;
+#else
+            new[]
+            {
+                CoreMetadataReference.SystemDataCommon
+            };
+#endif
+
+        internal static IEnumerable<MetadataReference> GetSystemDrawing() =>
+#if NETFRAMEWORK
+            FrameworkMetadataReference.SystemDrawing;
+#else
+            NuGetMetadataReference.SystemDrawingCommon();
+#endif
+
+        internal static IEnumerable<MetadataReference> GetSystemDirectoryServices() =>
+#if NETFRAMEWORK
+            FrameworkMetadataReference.SystemDirectoryServices;
+#else
+            NuGetMetadataReference.SystemDDirectoryServices();
+#endif
+
+        internal static IEnumerable<MetadataReference> GetSystemIoCompression() =>
+#if NETFRAMEWORK
+            FrameworkMetadataReference.SystemIOCompression
+                .Concat(FrameworkMetadataReference.SystemIOCompressionFileSystem);
+#else
+            new[]
+            {
+                CoreMetadataReference.SystemIoCompression,
+                CoreMetadataReference.SystemIoCompressionZipFile
+            };
+#endif
+
+        internal static IEnumerable<MetadataReference> GetSystemServiceModel() =>
+#if NETFRAMEWORK
+            FrameworkMetadataReference.SystemServiceModel;
+#else
+            NuGetMetadataReference.SystemPrivateServiceModel()
+                .Concat(NuGetMetadataReference.SystemServiceModelPrimitives());
+#endif
+
+        internal static IEnumerable<MetadataReference> GetSystemNetHttp() =>
+#if NETFRAMEWORK
+            FrameworkMetadataReference.SystemNetHttp;
+#else
+            new[]
+            {
+                CoreMetadataReference.SystemNetHttp,
+                CoreMetadataReference.SystemNetRequests,
+                CoreMetadataReference.SystemNetSecurity,
+                CoreMetadataReference.SystemNetServicePoint,
+                CoreMetadataReference.SystemNetSockets,
+                CoreMetadataReference.SystemNetPrimitives,
+                CoreMetadataReference.SystemNetWebClient
+            };
+#endif
+
+        internal static IEnumerable<MetadataReference> GetSystemSecurityCryptography() =>
+#if NETFRAMEWORK
+            FrameworkMetadataReference.SystemSecurityCryptographyAlgorithms;
+#else
+            new[]
+            {
+                CoreMetadataReference.SystemSecurityCryptographyAlgorithms,
+                CoreMetadataReference.SystemSecurityCryptographyX509Certificates,
+                CoreMetadataReference.SystemSecurityCryptographyCsp,
+                CoreMetadataReference.SystemSecurityCryptographyCng,
+                CoreMetadataReference.SystemSecurityCryptographyPrimitives
+            };
+#endif
+
+        internal static IEnumerable<MetadataReference> GetSystemSecurityPermissions() =>
+#if NETFRAMEWORK
+            Enumerable.Empty<MetadataReference>();
```

Comment:
please add comment "it's in mscorlib"

well... it's in many places, maybe comment at the top?

Reply:
nevermid this